cellxgene_schema_cli/cellxgene_schema/ontology.py

```diff
@@ -1,86 +1,201 @@
 """Methods for working with ontologies and the OLS."""
-from urllib.parse import quote_plus
+import os
+import gzip
+import json
+
+
+class geneChecker:
+    """Handles checking gene ids, retrieves symbols"""
+
+    GENE_FILES = {
+        "Homo sapiens": os.path.join(
+            os.path.dirname(os.path.realpath(__file__)),
+            "ontology_files/genes_homo_sapiens.csv.gz",
+        ),
+        "Mus musculus": os.path.join(
+            os.path.dirname(os.path.realpath(__file__)),
+            "ontology_files/genes_mus_musculus.csv.gz",
+        ),
+    }
+
+    def __init__(self, species):
+        """
+        :param str species: scientific name of a species, currently the only supported species are: "Homo sapiens"
+        and "Mus musculus"
+        """
+
+        if species not in self.GENE_FILES:
+            raise ValueError(f"{species} not supported.")
+
+        self.species = species
+        self.gene_dict = {}
+        with gzip.open(self.GENE_FILES[species], "rt") as genes:
+            for gene in genes:
+                gene = gene.rstrip().split(",")
+                self.gene_dict[gene[0]] = gene[1]
+
+    def is_valid_id(self, gene_id):
+        """
+        Returns True if the gene_id is a valid ENSEMBL id
+        """
+
+        return gene_id in self.gene_dict
+
+    def assert_gene_id(self, gene_id):
+        """
+        Raises error if the gene_id is not a valid ENSEBML id
+        """
+
+        if not self.is_valid_id(gene_id):
```

Comment:
Are you planning on further extending `is_valid_id`? Otherwise you can probably combine is_valid_id and assert_gene_id for now

Reply:
or you could just refactor `get_symbol` to catch the keyerror and raise a value error in `get_symbol`

Reply:
This will get used by the validator

Reply:
`assert_gene_id` will be used by the validator? or `is_valid_id` will be used by the validator? or `get_symbol`

Reply:
Oh sorry I mentioned in our call that `get_symbol` won't get use but it actually will.

Reply:
I removed `assert_gene_id` in 7f04114319441478d4cdbe553c344ba262f50303